cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1456,7 +1457,7 @@ makeShelleyKeyWitness (ShelleyTxBody txbody _) =
             vk        = getShelleyKeyWitnessVerificationKey sk
             signature = makeShelleySignature txhash sk
          in ShelleyKeyWitness $
-              Shelley.WitVKey vk signature
+              Shelley.WitVKey vk (coerce signature)
```

Comment:
I guess this is related to `EraIndependentTxBody`. In consensus, I typically use `eraIndTxBodyHash` + `makeWitnessVKey`

Reply:
I'd prefer to use something explicit, even if it's just a `castSignatureRole` that's defined in the ledger lib as `coerce` but constrained to the appropriate type.

Reply:
OK, I'll push an update

Reply:
@dcoutts Done